main.go

```diff
@@ -1,51 +1,166 @@
 package main
 
 import (
+	"bufio"
 	"encoding/json"
 	"fmt"
-	"io/ioutil"
+	"io"
 	"os"
 
 	"github.com/yazgazan/jaydiff/diff"
 )
 
 const (
-	statusUsage          = 2
-	statusReadError      = 3
-	statusUnmarshalError = 4
-	statusDiffError      = 5
-	statusDiffMismatch   = 6
+	statusUsage           = 2
+	statusReadError       = 3
+	statusUnexpectedError = 4
+	statusDiffError       = 5
+	statusDiffMismatch    = 6
 )
 
 var (
 	// Version is replaced by the tag when creating a new release
 	Version = "dev"
 )
+type nextJson func() (interface{}, int, error)
 
 func main() {
-	var err error
+	var (
+		lhs_file,rhs_file *os.File
+		err error
+	)
+
 	conf := readConfig()
 
-	lhs := parseFile(conf.Files.LHS)
-	rhs := parseFile(conf.Files.RHS)
+	// Open files for reading
+	lhs_file, err = os.Open(conf.Files.LHS)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: cannot read %s: %s\n", conf.Files.LHS, err.Error())
+		os.Exit(statusReadError)
+	}
+	defer lhs_file.Close()
+	rhs_file, err = os.Open(conf.Files.RHS)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: cannot read %s: %s\n", conf.Files.RHS, err.Error())
+		os.Exit(statusReadError)
+	}
+	defer rhs_file.Close()
+
+	//
+	if conf.JsonLines {
+		os.Exit(compareLoop(getNextJsonByLine(lhs_file), getNextJsonByLine(rhs_file),&conf))
+	} else {
+		os.Exit(compareLoop(getNextJson(lhs_file), getNextJson(rhs_file),&conf))
```

Comment:
Plain, single-valued, JSON files and JSON stream should not be handled the same way. jaydiff should be returning an error if it is expecting a single value but gets a JSON stream instead.